components/NativeApp/MessageSync.js

```diff
@@ -108,7 +108,9 @@ const MessageSync = ({ upsertDevice, me, client }) => {
       } else if (content.type === 'authorization') {
         checkPendingAppSignIn()
       } else if (content.type === 'push-route') {
-        Router.pushRoute(content.url.replace(PUBLIC_BASE_URL, ''))
+        Router.pushRoute(content.url.replace(PUBLIC_BASE_URL, '')).then(() => {
+          window.scrollTo(0, 0)
+        })
```

Comment:
There are potential `scrollTo` conflicts, to work around this we should back-off when there is a hash in the url:
```suggestion
        const targetUrl = content.url.replace(PUBLIC_BASE_URL, '')
        Router.pushRoute(targetUrl).then(() => {
          if (targetUrl.indexOf('#') === -1) {
            window.scrollTo(0, 0)
          }
        })
```

We should test opening `https://www.republik.ch/konto#newsletter`—used in newsletters—after deploy. Currently works on iOS most of the time.

I also just noticed that next 10 has already improved its scroll logic and seems to do scroll 0,0 by default now. But probably won't get around to do the update until late march (needs routing testing and potentially changes).